src/handlers/Firefox110.ts

```diff
@@ -0,0 +1,1041 @@
+
+		const transceiver = this._mapMidTransceiver.get(localId);
+
+		if (!transceiver)
+			throw new Error('associated RTCRtpTransceiver not found');
+
+		transceiver.direction = 'inactive';
+		this._remoteSdp!.pauseMediaSection(localId);
+
+		const offer = await this._pc.createOffer();
+
+		logger.debug(
+			'pauseSending() | calling pc.setLocalDescription() [offer:%o]',
+			offer);
+
+		await this._pc.setLocalDescription(offer);
+
+		const answer = { type: 'answer', sdp: this._remoteSdp!.getSdp() };
+
+		logger.debug(
+			'pauseSending() | calling pc.setRemoteDescription() [answer:%o]',
+			answer);
+
+		await this._pc.setRemoteDescription(answer);
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	async resumeSending(localId: string): Promise<void>
+	{
+		this.assertSendDirection();
+
+		logger.debug('resumeSending() [localId:%s]', localId);
+
+		const transceiver = this._mapMidTransceiver.get(localId);
+
+		if (!transceiver)
+			throw new Error('associated RTCRtpTransceiver not found');
+
+		transceiver.direction = 'sendonly';
+		this._remoteSdp!.resumeSendingMediaSection(localId);
+
+		const offer = await this._pc.createOffer();
+
+		logger.debug(
+			'resumeSending() | calling pc.setLocalDescription() [offer:%o]',
+			offer);
+
+		await this._pc.setLocalDescription(offer);
+
+		const answer = { type: 'answer', sdp: this._remoteSdp!.getSdp() };
+
+		logger.debug(
+			'resumeSending() | calling pc.setRemoteDescription() [answer:%o]',
+			answer);
+
+		await this._pc.setRemoteDescription(answer);
+	}
+
+	async replaceTrack(
+		localId: string, track: MediaStreamTrack | null
+	): Promise<void>
+	{
+		this.assertSendDirection();
+
+		if (track)
+		{
+			logger.debug(
+				'replaceTrack() [localId:%s, track.id:%s]', localId, track.id);
+		}
+		else
+		{
+			logger.debug('replaceTrack() [localId:%s, no track]', localId);
+		}
+
+		const transceiver = this._mapMidTransceiver.get(localId);
+
+		if (!transceiver)
+			throw new Error('associated RTCRtpTransceiver not found');
+
+		await transceiver.sender.replaceTrack(track);
+	}
+
+	async setMaxSpatialLayer(localId: string, spatialLayer: number): Promise<void>
+	{
+		this.assertSendDirection();
+
+		logger.debug(
+			'setMaxSpatialLayer() [localId:%s, spatialLayer:%s]',
+			localId, spatialLayer);
+
+		const transceiver = this._mapMidTransceiver.get(localId);
+
+		if (!transceiver)
+			throw new Error('associated transceiver not found');
+
+		const parameters = transceiver.sender.getParameters();
+
+		// NOTE: We require encodings given from low to high, however Firefox
+		// requires them in reverse order, so do magic here.
+		spatialLayer = parameters.encodings.length - 1 - spatialLayer;
+
+		parameters.encodings.forEach((encoding: RTCRtpEncodingParameters, idx: number) =>
+		{
+			if (idx >= spatialLayer)
+				encoding.active = true;
+			else
+				encoding.active = false;
+		});
+
+		await transceiver.sender.setParameters(parameters);
```

Comment:
I believe after this line, we should do what mediasoup has done in this [commit](https://github.com/versatica/mediasoup-client/commit/9000832d3b2193925c9492882a7a0ef230bbdc86), to persist the changes in sending encoding:

```suggestion
		await transceiver.sender.setParameters(parameters);

		this._remoteSdp!.muxMediaSectionSimulcast(localId, parameters.encodings);

		const offer = await this._pc.createOffer();

		logger.debug(
			'setMaxSpatialLayer() | calling pc.setLocalDescription() [offer:%o]',
			offer);

		await this._pc.setLocalDescription(offer);

		const answer = { type: 'answer', sdp: this._remoteSdp!.getSdp() };

		logger.debug(
			'setMaxSpatialLayer() | calling pc.setRemoteDescription() [answer:%o]',
			answer);

		await this._pc.setRemoteDescription(answer);
```

Reply:
We have already faced an issue with Chrome in the past, where after we start to screen share or send a custom track, the bandwidth was not respected anymore. 

And these changes from mediasoup fix this kind of behavior.